### PR TITLE
Replace `SafeLocalAllocHandle` in `System.Diagnostics.PerformanceCounter`

### PR DESCRIPTION
**Pull request by xtqqczze**
Follow-up #82411

We don't need a `SafeLocalAllocHandle` since we own the handle and immediately call `DangerousGetHandle`.

Additionally change type of `Interop+Kernel32+SECURITY_ATTRIBUTES.lpSecurityDescriptor` from `IntPtr` to `void*` which cleans up the code.

cc: @stephentoub

### CODE REVIEW

File: src/libraries/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/SharedPerformanceCounter.cs [hunk 1]
```diff
@@ -1756,7 +1761,7 @@ private unsafe void Initialize(string fileMappingName, int fileMappingSize, int
                 }
                 finally
                 {
-                    securityDescriptorPointer?.Close();
+                    Interop.Kernel32.LocalFree((IntPtr)pSecurityDescriptor);
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/SharedPerformanceCounter.cs:1764]: SafeLocalAllocHandle was using Marshal.FreeHGlobal.  Let's keep using it here.

File: src/libraries/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj [hunk 2]
```diff
@@ -93,6 +93,8 @@ System.Diagnostics.PerformanceCounter</PackageDescription>
              Link="Common\Interop\Windows\Kernel32\Interop.HandleOptions.cs" />
     <Compile Include="$(CommonPath)Interop\Windows\Kernel32\Interop.LoadLibrary.cs"
              Link="Common\Interop\Windows\Kernel32\Interop.LoadLibrary.cs" />
+    <Compile Include="$(CommonPath)Interop\Windows\Kernel32\Interop.LocalAlloc.cs"
+             Link="Common\Interop\Windows\Kernel32\Interop.LocalAlloc.cs" />
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj:97]: Once you switch back to Marshal.FreeHGlobal you can remove this.

**Review by stephentoub — APPROVED** [review]
LGTM other than two comments.

File: src/libraries/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/SharedPerformanceCounter.cs (hunk 1 above)
Comment (xtqqczze) [reply src/libraries/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/SharedPerformanceCounter.cs:1764]: `pSecurityDescriptor` is assigned by `ConvertStringSecurityDescriptorToSecurityDescriptor`, for which documentation states:
> To free the returned buffer, call the [LocalFree](https://learn.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-localfree) function.

If there's a reason to prefer an indirect call through `Marshal.FreeHGlobal`, I'll make the changes.

Comment (stephentoub) [reply src/libraries/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/SharedPerformanceCounter.cs:1764]: Simply that a) it's what was being used before and b) we don't need to add another P/Invoke to this library and can just call the public API.

Comment (xtqqczze) [reply src/libraries/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/SharedPerformanceCounter.cs]: Done.

**Review by stephentoub — APPROVED** [review]
Thanks.

**Pull request merged**